### PR TITLE
test: send xray reports

### PR DESCRIPTION
**Pull request by chriswldenyer**
Send reports to Jira xray.
Specify which test plan in the Jenkinsfile.
Currently implemented for on-demand and nightly tests.
Updated one readme.

### CODE REVIEW

**Comment by chriswldenyer** [comment]
bors try

**Comment by chriswldenyer** [comment]
bors try

**Comment by blaisedias** [comment]
Would it make more sense to have a utility function GetReporters(reportName) which creates the set of reporters for the e2e tests instead of constructing the required set of reporters in each test?
That way we can easily change the reporting method(s) without having to change all the test files.

File: test/e2e/common/junit/junit.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package junit
+
+import (
+	"os"
+)
+
+func ConstructJunitFileName(name string) string {
+	reportDir := os.Getenv("e2e_reports_dir")
```
Comment (arne-rusek) [inline test/e2e/common/junit/junit.go:8]: Please check that `e2e_reports_dir` is not empty and called hasn't forgotten to set it. I'd say `panic()` is good enough here?

File: test/e2e/common/junit/junit.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	testGroupPrefix := "e2e."
+	return reportDir + "/" + testGroupPrefix + name
```
Comment (arne-rusek) [inline test/e2e/common/junit/junit.go:10]: I second @blaisedias 's comment and would return fully constructed `[]Reporters` here. More DRY, better maintainability going forward.

**Review by arne-rusek — APPROVED** [review]
Two comments, overall LGTM

File: test/e2e/common/junit/junit.go (hunk 1 above)
Comment (chriswldenyer) [reply test/e2e/common/junit/junit.go]: Ok, check now added.

File: test/e2e/common/junit/junit.go (hunk 2 above)
Comment (chriswldenyer) [reply test/e2e/common/junit/junit.go]: Now done.

**Review by blaisedias — APPROVED** [review]
LGTM

File: test/e2e/common/reporter/junit.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/reporters"
+)
+
+func GetReporters(name string) []Reporter {
+	reportDir := os.Getenv("e2e_reports_dir")
+	if reportDir == "" {
+		panic("reportDir not defined")
```
Comment (arne-rusek) [inline test/e2e/common/reporter/junit.go:13]:
```suggestion
		panic("reportDir not defined - define via e2e_reports_dir environment variable")
```

reportDir is internal representation, for user-experience it's better to notice what needs to be changed w/o llooking into the code.

**Review by arne-rusek — APPROVED** [review]
One comment, otherwise LGTM

**Comment by chriswldenyer** [comment]
bors merge

**Comment by chriswldenyer** [comment]
bors try

**Comment by chriswldenyer** [comment]
bors try

**Comment by chriswldenyer** [comment]
bors merge

**Pull request merged**